internal/jobcontainers/jobcontainer.go

```diff
@@ -0,0 +1,393 @@
+// is mounted this call will just return nil.
+func (c *JobContainer) Release(ctx context.Context) error {
+	if c.sandboxMount != "" {
+		if err := removeSandboxMountPoint(ctx, c.sandboxMount); err != nil {
+			log.G(ctx).WithFields(logrus.Fields{
+				"id":   c.id,
+				"path": c.sandboxMount,
+			}).Warn("failed to remove sandbox volume mount")
+			return errors.Wrap(err, "failed to remove sandbox volume mount path")
+		}
+		if err := layers.UnmountContainerLayers(ctx, c.spec.Windows.LayerFolders, "", nil, layers.UnmountOperationAll); err != nil {
+			log.G(ctx).WithFields(logrus.Fields{
+				"id":    c.id,
+				"paths": c.spec.Windows.LayerFolders,
+			}).Warn("failed to unmount container layers")
+			return errors.Wrap(err, "failed to unmount container layers")
+		}
+		c.sandboxMount = ""
+	}
+	return nil
+}
+
+// Start starts the container. This is a noop as there isn't anything to "start".
+func (c *JobContainer) Start(ctx context.Context) error {
+	return nil
+}
+
+// Close closes any open handles.
+func (c *JobContainer) Close() error {
+	if err := c.job.Close(); err != nil {
+		return err
+	}
+	c.closedWaitOnce.Do(func() {
+		c.waitError = hcs.ErrAlreadyClosed
+		close(c.waitBlock)
+	})
+	return nil
+}
+
+// ID returns the ID of the container. This is the name used to create the job object.
+func (c *JobContainer) ID() string {
+	return c.id
+}
+
+// Shutdown gracefully shuts down the container.
+func (c *JobContainer) Shutdown(ctx context.Context) error {
+	log.G(ctx).WithField("id", c.id).Debug("shutting down job container")
+
+	return c.shutdown(ctx)
+}
+
+// shutdown will loop through all the pids in the container and send a signal to exit.
+// If there are no processes in the container it will just early return nil.
+// If the all processes exited message is not received within the context timeout set, it will
+// terminate the job.
+func (c *JobContainer) shutdown(ctx context.Context) error {
+	pids, err := c.job.Pids()
+	if err != nil {
+		return errors.Wrap(err, "failed to get pids in container")
+	}
+
+	if len(pids) == 0 {
+		return nil
+	}
+
+	for _, pid := range pids {
+		if err := windows.GenerateConsoleCtrlEvent(windows.CTRL_BREAK_EVENT, pid); err != nil {
+			return err
+		}
+	}
+
+	select {
+	case <-c.exited:
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+	return nil
+}
+
+// PropertiesV2 is not implemented for job containers. This is just to satisfy the cow.Container interface.
+func (c *JobContainer) PropertiesV2(ctx context.Context, types ...hcsschema.PropertyType) (*hcsschema.Properties, error) {
+	return nil, errors.New("propertiesV2 call not implemented for job containers")
+}
+
+// Properties is not implemented for job containers. This is just to satisfy the cow.Container interface.
+func (c *JobContainer) Properties(ctx context.Context, types ...schema1.PropertyType) (*schema1.ContainerProperties, error) {
+	return nil, errors.New("properties call not implemented for job containers")
+}
```

Comment:
Do we plan to mimic support for these?

Reply:
Yes, this is in another branch and forgot to include in this -_-. Will add!